File: lib/windows-phone-8/plugins/www/plugins/PGSocialShare/PGSocialShare.cs
using System;
using System.Runtime.Serialization;
using Microsoft.Phone.Tasks;

namespace WPCordovaClassLib.Cordova.Commands
{

    public class PGSocialShare : BaseCommand
    {
        public enum ShareType
        {
            Status = 0, // status update to twitter/facebook/... all supported device accounts
            Link        // share a link on twitter/facebook/ ... all supported device accounts
        }


        [DataContract]
        public class ShareOptions
        {
            [DataMember]
            public string url;

            [DataMember]
            public string title;

            [DataMember]
            public string message;

            [DataMember]
            public ShareType shareType = ShareType.Status; // default
        }

        public void share(string options)
        {
            ShareOptions opts = JSON.JsonHelper.Deserialize<ShareOptions>(options);
            switch (opts.shareType)
            {
                case ShareType.Status :
                    shareStatus(opts.message);
                    break;
                case ShareType.Link :
                    shareLink(opts.title, opts.url, opts.message);
                    break;
            }
        }

        protected void shareStatus(string msg)
        {
            ShareStatusTask shareStatusTask = new ShareStatusTask();
            shareStatusTask.Status = msg;
            shareStatusTask.Show();

            this.DispatchCommandResult();
        }

        protected void shareLink(string title, string url, string msg)
        {
            ShareLinkTask shareLinkTask = new ShareLinkTask();
            shareLinkTask.Title = title;
            shareLinkTask.LinkUri = new Uri(url, UriKind.Absolute);
            shareLinkTask.Message = msg;
            shareLinkTask.Show();

            this.DispatchCommandResult();
        }
    }
}

File: lib/windows-phone-8/plugins/www/plugins/PGSocialShare/PGSocialShare.js
/* MIT licensed */
// (c) 2011 Jesse MacFadyen,  Adobe Systems Incorporated



(function(window){

    var cdv = window.cordova || window.Cordova;

    navigator.plugins.pgSocialShare =
    {
        shareStatus:function(msg)
        {
            var options = {"message":msg,"shareType":0}; // 0 == status
            cdv.exec(null,null,"PGSocialShare","share", options);
        },

        shareLink:function(title,url,msg)
        {
            var options = {"message":msg, "title":title, "url":url, "shareType":1}; // 1 == link
            cdv.exec(null,null,"PGSocialShare","share", options);
        }
    }

})(window);